aicsimageio/readers/mrc_reader.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from typing import Any, Tuple
+
+import mrcfile
+import numpy as np
+import dask.array as da
+import xarray as xr
+from numpy.lib.recfunctions import structured_to_unstructured
+from fsspec.spec import AbstractFileSystem
+from fsspec.implementations.local import LocalFileSystem
+
+from .. import exceptions, types
+from ..utils import io_utils
+from ..dimensions import DimensionNames
+from .reader import Reader
+
+
+class MrcReader(Reader):
+    """
+    Read MRC files using mrcfile.
+
+    Parameters
+    ----------
+    image: types.PathLike
+        Path to image file to construct Reader for.
+    """
+
+    @staticmethod
+    def _is_supported_image(fs: AbstractFileSystem, path: str, **kwargs: Any) -> bool:
+        try:
+            with mrcfile.open(path, permissive=True, header_only=True):
+                return True
+        except (ValueError, TypeError):
+            return False
+
+    def __init__(self, image: types.PathLike):
+        # Expand details of provided image
+        self._fs, self._path = io_utils.pathlike_to_fs(image, enforce_exists=True)
+        # Catch non-local file system
+        if not isinstance(self._fs, LocalFileSystem):
+            raise NotImplementedError(
+                f"mrc reader not yet implemented for non-local file system. "
+                f"Received URI: {self._path}, which points to {type(self._fs)}."
+            )
+
+        if not self._is_supported_image(self._fs, self._path):
+            raise exceptions.UnsupportedFileFormatError(
+                self.__class__.__name__, self._path
+            )
+
+    @property
+    def scenes(self) -> Tuple[str, ...]:
+        if self._scenes is None:
+            # need full mmap open here because `is_image_stack` needs access to data shape
+            with mrcfile.mmap(self._path, permissive=True) as mrc:
+                if mrc.is_image_stack():
+                    n_scenes = mrc.header.nz
+                else:
+                    n_scenes = mrc.header.nz / mrc.header.mz
+                self._scenes = [f'Image:{i}' for i in range(int(n_scenes))]
```

Comment:
Nit, can you use our `generate_ome_image_id` function: https://github.com/AllenCellModeling/aicsimageio/blob/main/aicsimageio/metadata/utils.py#L104

its doing exactly that same f-string but if we (or OME) ever change that standard we can clean it up in a single location rather than multiple.

Reply:
Done!